README.md

```diff
@@ -24,39 +26,63 @@ To install Go, visit [this link](https://golang.org/dl/).
 ## Usage
 Before using MSAL Go, you will need to [register your application with the Microsoft identity platform](https://docs.microsoft.com/azure/active-directory/develop/quickstart-v2-register-an-app).
 
+### Public Surface
+
+The Public API of the library can be found in the following directories under `apps`.
+
+```
+apps/ - Contains all our code
+  confidential/ - The confidential application API
+  public/ - The public application API
+  cache/ - The cache interface that can be implemented to provide persistence cache storage of credentials
+```
+
 Acquiring tokens with MSAL Go follows this general three step pattern. There might be some slight differences for other token acquisition flows. Here is a basic example:
+
 1. MSAL separates [public and confidential client applications](https://tools.ietf.org/html/rfc6749#section-2.1). So, you would create an instance of a `PublicClientApplication` and `ConfidentialClientApplication` and use this throughout the lifetime of your application.
-```go
-publicClientApp, err := msal.CreatePublicClientApplication("your_client_id",
-    "https://login.microsoftonline.com/Enter_the_Tenant_Name_Here")
-```
+   * Initializing a public client:
 
-2. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
-```go
-accounts := publicClientApp.GetAccounts()
-// Assuming the user wanted the first account
-userAccount := accounts[0]
-scopes := []string{"your_scope"}
-silentParams := msal.CreateAcquireTokenSilentParametersWithAccount(scopes, userAccount)
-result, err := publicClientApp.AcquireTokenSilent(silentParams)
-```
+    ```go
+    app, err := public.New("client_id", public.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+   * Initializing a confidential client:
+
+    ```go
+    // Initializing the client credential
+    cred, err := confidential.NewCredFromSecret("client_secret")
+    if err != nil {
+        return nil, fmt.Errorf("could not create a cred from a secret: %w", err)
+    }
+    app, err := confidential.New("client_id", cred, confidential.WithAuthority("https://login.microsoftonline.com/Enter_The_Tenant_Name_Here"))
+    ```
+
+1. MSAL comes packaged with an in-memory cache. Utilizing the cache is optional, but we would highly recommend it.
+
+    ```go
+    var userAccount public.Account
+    accounts := app.Accounts()
```

Comment:
what is accounts, and why get a userAccount? is this code even possible? it doesn't look right to me